### PR TITLE
Ft sign up

### PR DESCRIPTION
**Pull request by thoth-ky**
- Complete working signup

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,4 +1,26 @@
...
+  end
+
   def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      flash[:success] = "Welcome to the Sample App!"
```
Comment (chalchuck) [inline app/controllers/users_controller.rb:14]: I am going to crucify you .... make the redirect and flash a one liner.

File: app/helpers/users_helper.rb [hunk 2]
```diff
@@ -1,2 +1,9 @@
 module UsersHelper
+    # Returns the Gravatar for the given user.
+  def gravatar_for(user, options = { size: 80 })
```
Comment (chalchuck) [inline app/helpers/users_helper.rb]: This method does more than one task, make it adhere to Single Responsibility Principle

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<% provide(:title, @user.name) %>
+<div class="row">
+  <aside class="col-md-4">
+    <section class="user_info">
+      <h1>
+        <%= gravatar_for @user %>
```
Comment (chalchuck) [inline app/views/users/show.html.erb]: Prefer to use () when calling a method with parameters

**Review by chalchuck — COMMENTED** [review]
Please review and revert.
I Hope you have configured Auto deployments to Heroku

File: config/puma.rb [hunk 4]
```diff
@@ -1,47 +1,16 @@
...
-# or connections that may have been created at application boot, Ruby
-# cannot share connections between processes.
-#
-# on_worker_boot do
-#   ActiveRecord::Base.establish_connection if defined?(ActiveRecord)
-# end
-
-# Allow puma to be restarted by `rails restart` command.
-plugin :tmp_restart
+on_worker_boot do
```
Comment (chalchuck) [inline config/puma.rb:11]: No Comments!!!!!!!!!

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (thoth-ky) [reply app/controllers/users_controller.rb:14]: Using it this way gives me an easy way to leverage bootstrap alert classes like here:
https://stackoverflow.com/a/46594462

File: app/helpers/users_helper.rb [hunk 5]
```diff
@@ -1,9 +1,9 @@
 module UsersHelper
     # Returns the Gravatar for the given user.
-  def gravatar_for(user, options = { size: 80 })
+  def gravatar_for(user, size: 80)
     gravatar_id = Digest::MD5::hexdigest(user.email.downcase)
-    size = options[:size]
-    gravatar_url = "https://secure.gravatar.com/avatar/#{gravatar_id}?s=#{size}"
-    image_tag(gravatar_url, alt: user.name, class: "gravatar")
+    image_tag("https://secure.gravatar.com/avatar/#{gravatar_id}?s=#{size}",
```
Comment (chalchuck) [inline app/helpers/users_helper.rb:5]: This method still does not adhere to Single Responsibility.

**Review by chalchuck — APPROVED** [review]
@jmutuku95 Good to merge!

PS: Read on Single Responsibility and try using it in all the code you write.

**Review by chalchuck — COMMENTED** [review]
good to merge

**Comment by thoth-ky** [comment]
This PR was created to merge to master by mistake. It's changes were carried to develop by `FT Basic Login` Therefore it is pointless to merge it.

**Pull request closed without merging**